packages/ws/src/ws/WebSocketManager.ts

```diff
@@ -0,0 +1,269 @@
+import type { REST } from '@discordjs/rest';
+import { AsyncEventEmitter } from '@vladfrangu/async_event_emitter';
+import {
+	APIGatewayBotInfo,
+	GatewayIdentifyProperties,
+	GatewayPresenceUpdateData,
+	RESTGetAPIGatewayBotResult,
+	Routes,
+} from 'discord-api-types/v10';
+import type { WebSocketShardDestroyOptions, WebSocketShardEventsMap } from './WebSocketShard';
+import type { IShardingStrategy } from '../strategies/sharding/IShardingStrategy';
+import { SimpleShardingStrategy } from '../strategies/sharding/SimpleShardingStrategy';
+import { CompressionMethod, DefaultWebSocketManagerOptions, Encoding } from '../utils/constants';
+import { Awaitable, range } from '../utils/utils';
+
+/**
+ * Represents a range of shard ids
+ */
+export interface ShardRange {
+	start: number;
+	end: number;
+}
+
+/**
+ * Session information for a given shard, used to resume a session
+ */
+export interface SessionInfo {
+	/**
+	 * Session id for this shard
+	 */
+	sessionId: string;
+	/**
+	 * The sequence number of the last message sent by the shard
+	 */
+	sequence: number;
+	/**
+	 * The id of the shard
+	 */
+	shardId: number;
+	/**
+	 * The total number of shards at the time of this shard identifying
+	 */
+	shardCount: number;
+}
+
+/**
+ * Required options for the WebSocketManager
+ */
+export interface RequiredWebsSocketManagerOptions {
+	/**
+	 * The token to use for identifying with the gateway
+	 */
+	token: string;
+	/**
+	 * The intents to request
+	 */
+	// Wonder if there's a better abstraction that could be done here?
+	intents: number;
```

Comment:
Gateway intents are still an enum so you can use -types' enum for now

Reply:
oh, didn't realize that was a thing, ty!